### PR TITLE
Feat(web): Style password toggle #DS-192

### PR DESCRIPTION
**Pull request by crishpeen**

### CODE REVIEW

**Comment by literat** [comment]
Screenshot 2022-04-06 at 16 49 01

It is possible to avoid conflict with browser plugins? In this case Lastpass. But maybe we will have a conflict with other password generators.

**Comment by crishpeen** [comment]
> It is possible to avoid conflict with browser plugins? In this case Lastpass. But maybe we will have a conflict with other password generators.

It should be fixed now. Can you try it please?

**Comment by literat** [comment]
Screenshot 2022-04-07 at 21 30 17
Screenshot 2022-04-07 at 21 31 02

Ok, well played 👍 Much better I think. Thanks ❤️

**Review by literat — APPROVED** [review]
LGTM

File: examples/web/index.html [hunk 1]
```diff
@@ -220,11 +257,31 @@ <h2 class="typography-heading-medium-text mt-1100 mb-800">TextField</h2>
...
+                <div class="TextField__password-toggle">
+                  <input type="password" id="textfieldInlinePassword" class="TextField__input" value="Password" />
+                  <button type="button" class="TextField__password-toggle__button" role="switch" aria-pressed="false" aria-label="Show password" data-toggle="password">
+                    <span class="TextField__password-toggle__icon" aria-hidden="true"></span>
+                  </button>
+                </div>
               </div>
               <button type="button" class="Button Button--primary">Button</button>
             </div>
+            <script type="text/javascript">
```
Comment (adamkudrna) [inline examples/web/index.html:270]:
```suggestion
            <script>
```

File: examples/web/index.html [hunk 2]
```diff
@@ -220,11 +257,31 @@ <h2 class="typography-heading-medium-text mt-1100 mb-800">TextField</h2>
...
+              const handlePasswordToggleClick = (event) => {
```
Comment (adamkudrna) [inline examples/web/index.html:271]: ⚠️ This needs to be refactored once #250 is merged.

File: packages/web/src/components/TextField/_TextField.scss [hunk 3]
```diff
@@ -46,16 +49,50 @@
     width: 100%;
 }
 
-.TextField:hover > .TextField__input,
-.TextField__input:hover {
-    border-color: theme.$input-border-color-hover;
+.TextField__password-toggle {
```
Comment (adamkudrna) [inline packages/web/src/components/TextField/_TextField.scss:52]:
```suggestion
.TextField__passwordToggle {
```

File: packages/web/src/components/TextField/_TextField.scss [hunk 4]
```diff
@@ -46,16 +49,50 @@
...
+
+.TextField__password-toggle__button {
+    @include reset.button();
+
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    padding: theme.$input-password-toggle-padding;
+
+    &::before {
```
Comment (adamkudrna) [inline packages/web/src/components/TextField/_TextField.scss:71]: Would you maybe drop a comment what the pseudo-element stands for?

File: packages/web/src/components/TextField/_TextField.scss [hunk 5]
```diff
@@ -46,16 +49,50 @@
...
+        right: 0;
+        bottom: 0;
+        left: 0;
+        border: theme.$input-border-width theme.$input-border-style theme.$input-border-color-default;
+        border-radius: theme.$input-border-radius;
+        pointer-events: none;
+    }
+}
+
+.TextField__password-toggle__icon {
```
Comment (adamkudrna) [inline packages/web/src/components/TextField/_TextField.scss:84]: Maybe the time for Sass icon abstraction has come? It can be surely done in a separate PR.

File: packages/web/src/components/TextField/_TextField.scss [hunk 6]
```diff
@@ -77,21 +116,57 @@
...
-    border-color: theme.$input-border-color-disabled;
 
     &::placeholder {
         color: theme.$input-placeholder-color-disabled;
     }
 }
 
+.TextField > .TextField__input:disabled,
+.TextField--disabled > .TextField__input,
+.TextField .TextField__password-toggle .TextField__input:disabled ~ .TextField__password-toggle__button::before,
```
Comment (adamkudrna) [inline packages/web/src/components/TextField/_TextField.scss:146]: How simple! 🤷🏻‍♂️ 😄

File: packages/web/src/components/TextField/README.md [hunk 7]
```diff
@@ -38,3 +38,32 @@ and show if the input is required.
   <div class="TextField__message">Message</div>
 </div>
 ```
+
+## TextField password toggle
+
+TextField with `type="password"` can have a toggle button. When toggling don't forget to change
+not only the input type but also `aria-pressed` and `aria-label`.
```
Comment (adamkudrna) [inline packages/web/src/components/TextField/README.md:45]: > When toggling don't forget to change not only the input type but also `aria-pressed` and `aria-label`.

This needs to be adjusted once #250 is merged.

**Review by adamkudrna — CHANGES_REQUESTED** [review]
Great job! 👍🏻 

First comments are here, but I'd like to see it once more when it's rebased after #250 has been merged.

File: examples/web/index.html (hunk 2 above)
Comment (crishpeen) [reply examples/web/index.html:271]: Then I should merge this first.

File: packages/web/src/components/TextField/_TextField.scss (hunk 6 above)
Comment (crishpeen) [reply packages/web/src/components/TextField/_TextField.scss:146]: 🤕

File: packages/web/src/components/TextField/_TextField.scss (hunk 5 above)
Comment (crishpeen) [reply packages/web/src/components/TextField/_TextField.scss:84]: I will create an issue.

**Review by adamkudrna — APPROVED** [review]
